From 911c241d6d0d569074ecf05136dfd5cb9694876e Mon Sep 17 00:00:00 2001
From: Andreas Henriksson <andreas@fatal.se>
Date: Fri, 2 Jul 2010 09:41:10 +0200
Subject: [PATCH 6/7] Patch from upstream (rev 2521).

Disable dot stripping code since it's still broken
and noone has been able to figure it out (yet).
---
 libarchive/archive_read_support_format_iso9660.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/libarchive/archive_read_support_format_iso9660.c b/libarchive/archive_read_support_format_iso9660.c
index 2d3a855..8661532 100644
--- a/libarchive/archive_read_support_format_iso9660.c
+++ b/libarchive/archive_read_support_format_iso9660.c
@@ -1771,9 +1771,11 @@ parse_file_info(struct archive_read *a, struct file_info *parent,
 			*wp = L'\0';
 		}
 
+#if 0 /* XXX: this somehow manages to strip of single-character file extensions, like '.c'. */
 		/* Chop off trailing '.' from filenames. */
 		if (*(wp-1) == L'.')
 			*(--wp) = L'\0';
+#endif
 
 		/* store the result in the file name field. */
 		archive_strappend_w_utf8(&file->name, wbuff);
-- 
1.7.1

